server/recording_api.go

```diff
@@ -15,6 +15,57 @@ import (
 
 var callRecordingActionRE = regexp.MustCompile(`^\/calls\/([a-z0-9]+)/recording/(start|stop|publish)$`)
 
+const recordingJobStartTimeout = 15 * time.Second
+
+func (p *Plugin) recJobTimeoutChecker(callID, jobID string) {
+	time.Sleep(recordingJobStartTimeout)
+
+	state, err := p.kvGetChannelState(callID)
+	if err != nil {
+		p.LogError("failed to get channel state", "error", err.Error())
+		return
+	}
+
+	recState, err := state.getRecording()
+	if err != nil {
+		p.LogError("failed to get recording state", "error", err.Error())
+		return
+	}
+
+	// If the recording hasn't started (bot hasn't joined yet) we notify the
+	// client.
+	var clientState *RecordingStateClient
+	if recState.JobID == jobID && recState.StartAt == 0 {
+		if err := p.kvSetAtomicChannelState(callID, func(state *channelState) (*channelState, error) {
+			recordingState, err := state.getRecording()
+			if err != nil {
+				return nil, err
+			}
+			if recordingState.JobID != jobID {
+				return nil, fmt.Errorf("invalid recording job")
+			}
```

Comment:
would this ever happen? (multiple jobs started I guess?) -- but if it does, what happens to that invalid job? Should we do cleanup?

Reply:
This is purely to prevent a potential edge case due to a race condition. We check above that the job matches but by the time we read from K/V store the recording could have stopped and a new one started. At this point we don't want to stop the new one as the request was meant for the previous one.